lib/ui/barcode_format_item.dart

```diff
@@ -2,12 +2,12 @@ import 'package:flutter/material.dart';
 import 'package:barcode_scanner/barcode_scanning_data.dart';
 
 class BarcodeFormatItemWidget extends StatelessWidget {
-  BarcodeFormat format;
-  bool selected;
-  ValueChanged<bool> onSelect;
+  final BarcodeFormat format;
+  final bool selected;
+  final ValueChanged<bool?> onSelect;
```

Comment:
maybe its better to make bool not nullable?  i would rather make ! to make it crash if we not added some barcode type.

Reply:
It's how Flutter Checkbox defined the onChanged delegate:

`final ValueChanged<bool?>? onChanged;`

Which is kind of funny, since it plays with this variable:

```
/// Whether this checkbox is checked.
/// This property must not be null.
final bool? value;
```

I am wrapping it like this, because I don't like it either:

```
final ValueChanged<bool> onSelect;
ValueChanged<bool?>? _onSelect;
  
BarcodeFormatItemWidget(this.format, this.selected, {required this.onSelect}) {
    _onSelect = (value)=>{
        onSelect(value!)
    };
}
```

What do you think @4BooM04 ?